src/downloader.rs

```diff
@@ -36,34 +37,78 @@ impl Downloader {
             .build();
 
         // Download the files asynchronously.
-        let _tasks = stream::iter(downloads)
+        stream::iter(downloads)
             .map(|d| self.fetch(&client, d))
             .buffer_unordered(self.concurrent_downloads)
             .collect::<Vec<_>>()
-            .await;
+            .await
     }
 
     /// Fetches the files and write them to disk.
-    async fn fetch(&self, client: &ClientWithMiddleware, download: &Download) -> Result<(), Error> {
+    async fn fetch(&self, client: &ClientWithMiddleware, download: &Download) -> Summary {
+        // Create a download summary.
+        let mut summary = Summary::new(download.clone(), StatusCode::BAD_REQUEST, 0);
+
         // Request the file.
         debug!("Fetching {}", download.url.clone());
```

Comment:
```suggestion
        debug!("Fetching {}", &download.url);
```

Reply:
You could also keep a field called `url` here, this will allow you to filter your logs by specific keys using [tracing_subscriber::EnvFilter](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/struct.EnvFilter.html):
```suggestion
        debug!(url = ?download.url, "Fetching url");
```
`?` means implement the display method for `Url`